### PR TITLE
Refactor use of CliProfileManager 

### PR DESCRIPTION
**Pull request by JillieBeanSim**
## Proposed changes

Remove the use of CLIProfileManager where we can, in cases that we can't make sure to check for no existence of Config file.

## Release Notes

Milestone: 2.3.0

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

File: packages/zowe-explorer/src/extension.ts [hunk 1]
```diff
@@ -363,7 +363,7 @@ async function watchConfigProfile(context: vscode.ExtensionContext) {
             new vscode.RelativePattern(vscode.workspace.workspaceFolders[0].uri.fsPath, "zowe.config.user.json")
         );
 
-    context.subscriptions.push(globalProfileWatcher);
+    context.subscriptions.push(workspaceProfileWatcher);
     workspaceProfileWatcher && context.subscriptions.push(globalProfileWatcher);
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/extension.ts:367]: @kristinochka I didn't see it when testing your PR but did notice while working on another branch.  This had a double subscription on the same watcher and would result in a stack error in the debug console when deleting a config that was being watched in workspace.  This changed fixed that issue.

**Review by rudyflores — APPROVED** [review]
Re-tested refresh on v2 profiles and it seems to be working as intended. As for the contents of this PR, they seem to be working well too! Thanks @JillieBeanSim !

**Comment by kristinochka** [comment]
What specific area should be tested?

**Comment by JillieBeanSim** [comment]
> What specific area should be tested?
> 
> I noticed an issue where updating credentials via "Update Credentials" button breaks ZE, but seeing it on master as well.

This PR deals mostly with APIs so any code in ZE that calls them and issue TSO Command.  can see what changed with files changed tab

**Comment by JillieBeanSim** [comment]
> What specific area should be tested?
> 
> I noticed an issue where updating credentials via "Update Credentials" button breaks ZE, but seeing it on master as well.

Fix for this will be on #1891

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**